worlds/civ_6/Items.py

```diff
@@ -0,0 +1,296 @@
+            id_offset=id_base,
+            classification=progression,
+            progression_name=None,
+            civ_name=item_name
+        )
+        progressive_id_base += 1
+    return progressive_table
+
+
+def _generate_progressive_era_items(id_base: int) -> List[CivVIItemData]:
+    """Generates the single progressive district item"""
+    era_table = {}
+    # Generate progressive eras
+    progressive_era_name = format_item_name("PROGRESSIVE_ERA")
+    era_table[progressive_era_name] = CivVIItemData(
+        name=progressive_era_name,
+        civ_vi_id=0,
+        cost=0,
+        item_type=CivVICheckType.ERA,
+        id_offset=id_base,
+        classification=ItemClassification.progression,
+        progression_name=None,
+        civ_name="PROGRESSIVE_ERA"
+    )
+    return era_table
+
+
+def _generate_goody_hut_items(id_base: int) -> List[CivVIItemData]:
+    # Generate goody hut items
+    goody_huts = get_filler_item_data()
+    goody_table = {}
+    goody_base = 0
+    for value in goody_huts.values():
+        goody_table[value.name] = CivVIItemData(
+            name=value.name,
+            civ_vi_id=goody_base,
+            cost=0,
+            item_type=CivVICheckType.GOODY,
+            id_offset=id_base,
+            classification=ItemClassification.filler,
+            progression_name=None,
+            civ_name=value.civ_name
+        )
+        goody_base += 1
+    return goody_table
+
+
+def generate_item_table() -> Dict[str, CivVIItemData]:
+    era_required_items = get_era_required_items_data()
+    required_items: List[str] = []
+    for key, value in era_required_items.items():
+        required_items += value
+
+    progressive_items = get_flat_progressive_districts()
+
+    item_table = {}
+
+    def get_id_base():
+        return len(item_table.keys())
+
+    item_table = {**item_table, **_generate_tech_items(get_id_base(), required_items, progressive_items)}
+    item_table = {**item_table, **_generate_civics_items(get_id_base(), required_items, progressive_items)}
+    item_table = {**item_table, **_generate_progressive_district_items(get_id_base())}
+    item_table = {**item_table, **_generate_progressive_era_items(get_id_base())}
+    item_table = {**item_table, **_generate_goody_hut_items(get_id_base())}
+
+    return item_table
+
+
+def get_items_by_type(item_type: CivVICheckType, item_table: Dict[str, CivVIItemData]) -> List[CivVIItemData]:
+    """
+    Returns a list of items that match the given item type
+    """
+    return [item for item in item_table.values() if item.item_type == item_type]
+
+
+def get_random_filler_by_rarity(world: 'CivVIWorld', rarity: FillerItemRarity, item_table: Dict[str, CivVIItemData]) -> CivVIItemData:
+    """
+    Returns a random filler item by rarity
+    """
+    items = [item for item in get_filler_item_data().values() if item.rarity == rarity]
+    return items[world.random.randint(0, len(items) - 1)]

```

Comment:
```suggestion
def get_random_filler_by_rarity(world: 'CivVIWorld', rarity: FillerItemRarity) -> CivVIItemData:
    """
    Returns a random filler item by rarity
    """
    items = [item for item in get_filler_item_data().values() if item.rarity == rarity]
    return items[world.random.randint(0, len(items) - 1)]
```
item_table part of function isn't used
Return type doesn't match what's being returned -- might be fine here though, not something I'd block

Reply:
Removed, leftover from old implementation 👍